spec/controllers/lesson_completions_controller_spec.rb

```diff
@@ -14,67 +14,47 @@
     end
 
     it "DELETE #destroy unauthorized" do
-      delete :destroy, :lesson_id => lc_attrs[:lesson_id]
+      delete :destroy, lesson_id: lc_attrs[:lesson_id]
       expect(response).to have_http_status(:unauthorized)
     end
   end
 
   context "after authentication" do
     before do
-      sign_in(student)
+      allow(controller).to receive(:current_user).and_return(student)
+      allow(Lesson).to receive(:find).with(lc_attrs[:lesson_id]).
+          and_return(lesson)
     end
 
     describe "POST #create" do
-      context "with valid attributes" do
-        it "saves the new event to the database" do
-          expect{
-            post :create, lc_attrs
-          }.to change(LessonCompletion, :count).by(1)
-        end
-
-        it 'renders the template' do
+      it "saves the lesson_completion record to the database" do
+        expect{
           post :create, lc_attrs
-          expect(response).to render_template('create')
-        end
+        }.to change(LessonCompletion, :count).by(1)
       end
 
-      context "with invalid attributes" do
-        it "does not save the new event to the database" do
-          expect{
-            post :create, lc_invalid_attrs
-          }.to change(LessonCompletion, :count).by(0)
-        end
-
-        it "returns a bad request status" do
-          post :create, lc_invalid_attrs
-          expect(response).to have_http_status(:bad_request)
-        end
+      it 'renders the template' do
+        post :create, lc_attrs
+        expect(response).to render_template('create')
       end
     end
 
     describe "DELETE #destroy" do
-      context "without an existing lesson_completions object" do
-        it "returns a bad request status" do
-          delete :destroy, :lesson_id => lc_attrs[:lesson_id]
-          expect(response).to have_http_status(:bad_request)
-        end
-      end
-
-      context "with an existing lesson_completions object and valid request attributes" do
+      before do
+        lesson_completions = double('LessonCompletions')
+        allow(LessonCompletion).to receive(:where).with(lc_attrs).
+          and_return(lesson_completions)
 
-        let!(:lc){ LessonCompletion.create(lc_attrs) }
-
-        it "destroys the existing object" do
-          expect{
-            delete :destroy, :lesson_id => lc.lesson_id
-          }.to change(LessonCompletion, :count).by(-1)
-        end
-
-        it 'renders the template' do
-          delete :destroy, :lesson_id => lc.lesson_id
-          expect(response).to render_template('create')
-        end
+        allow(lesson_completions).to receive(:first).and_return(lesson_completion)
+        expect(lesson_completion).to receive(:destroy)
```

Comment:
This is a little bit off, generally we should avoid expects in a before block as it can lead to some unexpected problems. 

You can change this to  `allow(lesson_completion).to receive(:destroy)` to allow the lesson completion object to receive this method.